### ISSUE TITLE
Template: Author names getting weird

### ISSUE THREAD

**Issue by mguidoti** [gsautter/goldengate-imagine#833]
Hi Guido,

Check this out. 

I'm doing a Zoosystema template and while the authors are being correctly identified, on the metadata later on, a pattern (space & space) is added in between every and each char on the auhthors' names.

See screenshots:

![image](https://user-images.githubusercontent.com/44419136/71015679-c9422e00-20d2-11ea-8210-f2f774e73d32.png)

![image](https://user-images.githubusercontent.com/44419136/71015738-e24adf00-20d2-11ea-8768-5b25d1dbe5c0.png)

Here's the string: K & a & m & a & l & a & n & a & t & h & a & n &  & V & N & K & I &  & o & h & a & n & r & a & j &  & P & H & N & H

Do you have any idea what we did wrong?

Thanks,

**Comment by gsautter** [comment]
Considering the pattern you specified, small wonder ... just try and hit the "Test" button and see what it matches ... there are several points (see also https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html):
- inside square brackets, character sequences are not words, but _character classes_, meaning "either one of the following" or, with the negation, any character _but_ the following
- not even sure the "or" pipe symbol is valid inside square brackets
- the square brackets define a character class, and the whole expression matches any _one_ letter within this class ... to match a sequence of characters, you have to add a quantifier (`+` or `*`, or `{n,m}`) that allows matching multiple characters in sequence

To match a name, maybe try in parts ... starting with a first name (which apparently comes in title case): `[A-Z][a-z]+`
- `[A-Z]` matches any _one_ character from the A-Z range
- `[a-z]+` matches one or more characters from the a-z range
- `[A-Z][a-z]+` matches any sequence of the two, i.e., one A-Z character followed by one or more a-z characters

The last name is even easier (assuming there is no single-letter last names): `[A-Z]{2,}`, which matches two or more (`{2,}` is "two to unspecified") characters from the A-Z range

As a whole, with a space (I've come to always wildcard spaces) in between, this makes `[A-Z][a-z]+\s[A-Z]{2,}`, or, as I've come to always wildcard spaces, `[A-Z][a-z]+\s*[A-Z]{2,}`

**Comment by gsautter** [comment]
It is also possible to allow some sequence to occur multiple times (say, multiple first names, or a first name and a middle name, which is the same in terms of syntax): ([A-Z][a-z]+\s*)+[A-Z]{2,}
- the parenthesis makes `[A-Z][a-z]+\s*` a _capturing group_ that can repeat (or even be optional) as a whole
- the `+` specifies that capturing group (name in title case plus subsequent space(s)) can occur none or more times

**Comment by gsautter** [comment]
Also, you can add optional parts, like middle initials: `[A-Z][a-z]+\s*+([A-Z]\.\s*)*[A-Z]{2,}`
- `[A-Z]\.\s*` matches a single upper case character followed by a dot and maybe some space
- the parentheses make it a capturing group, like above
- `*` specifies that capturing group can occur zero or more times

**Comment by mguidoti** [comment]
Yes, we thought that after clicking on the test button and checking what was highlighted, that was enough to understand what was being 'caught'. That's why we didn't specify a more specific pattern for authors - but looking at the log I noticed that you're right. 

The challenge now is getting all 'é', 'ã', 'ü' and other strange characters. 

We'll work on this. 

Thanks, Guido

**Comment by gsautter** [comment]
No worries about the "strange" characters (they are called "diacritics") ... they normalize out to their base forms before matching, e.g. "é" to "e".
Diacritics are a pain in the back to handle in patterns, and thus I introduced said normalization years ago. It's the same underlying functionality as the "Normalize Characters" option in the XML View, by the way, only not optional and under the hood.

**Comment by mguidoti** [comment]
All right, you're right. Already managed to get a RegEx pattern working for authors' names. 

Thanks!
